strongback-executable/src/org/strongback/executable/newproject/NewProject.java

```diff
@@ -0,0 +1,259 @@
+        public static final String MISSING_STRONGBK = "Could not locate the strongback directory. Double check that the strongback"
+                                                      + " folder is in the same directory as your wpilib folder.";
+        public static final String SUCCESS          = "Successfully created new project at: ";
+        
+        public static final String HELP =  "usage: strongback newproject [options] -d <directory> -n <project_name>"
+                                    + LS + "       strongback newproject [options] -r <project_root>"
+                                    + LS + ""
+                                    + LS + "Description"
+                                    + LS + "  Utility to create a new FRC robot project compatible with the strongback library"
+                                    + LS + "  If -d and -n are specified, the project will be created in a new directory inside"
+                                    + LS + "  -d with the name -n. If -r is specified, the project will be created directly inside"
+                                    + LS + "  -r with the project name being the same as -r. No files will be overwritten unless -o"
+                                    + LS + "  is specified."
+                                    + LS + ""
+                                    + LS + "Options"
+                                    + LS + "  -d <parent_directory>"
+                                    + LS + "    The directory to create the new project directory in"
+                                    + LS + ""
+                                    + LS + "  -e"
+                                    + LS + "    Adds project metadata for the Eclipse Java IDE"
+                                    + LS + ""
+                                    + LS + "  -h"
+                                    + LS + "    Displays help information"
+                                    + LS + ""
+                                    + LS + "  -n <project_name>"
+                                    + LS + "    The name of the new project"
+                                    + LS + ""
+                                    + LS + "  -o"
+                                    + LS + "    Forces overwriting of exisiting files"
+                                    + LS + ""
+                                    + LS + "  -p <package_name>"
+                                    + LS + "     Specifies a custom initial package for Robot.java"
+                                    + LS + ""
+                                    + LS + "  -v"
+                                    + LS + "     Displays version information"
+                                    ;
+        
+        /* Version Information */
+        // TODO Can we update this with ant?
+        public static final String VERSION_HEAD = "Strongback New Project Utility";
+        public static final String VERSION      = "1.1.0 compiled on 1 Oct 2015";
```

Comment:
Yes. Basically, we need a properties file in the JAR (e.g., `org/strongback/strongback.properties`) that would look like:

```
version=${project.version}
buildDate=${project.version}
```

Ant can either [generate this](https://ant.apache.org/manual/Tasks/propertyfile.html) or append to an existing file in the source, or it can even replace these variables when it builds (see [here](http://stackoverflow.com/questions/4508122/replacing-all-tokens-based-on-properties-file-with-ant)).

Then, we add methods in `Strongback` to expose these values. For example, we could add this interface:

    public static interface Metadata {
        public String version();
        public String buildDate();
    }

and then add a method to the `Strongback` class that reads the properties file and returns the values. Really, `Metadata` just wraps that properties file using a `Properties` object:

    public static Metadata metadata() {
        InputStream file = Strongback.class.getResourceAsStream("org/strongback/strongback.properties");
        assert file != null;
        Properties props = Properties.load(file);
        return new Metadata() {
            public String version() {
                return props.getProperty("version")
            }
            public String buildDate() {
                return props.getProperty("buildDate")
            }
        };
    }

Maven makes it very easy to update these properties files with the Maven-specific properties (like `${project.version}`), but Ant should be fine since we'll likely have to specify the version in the build script anyway.

You can either implement this now, or keep your TODO and create an issue to fix it.